plugins/browserDetection.ts

```diff
@@ -0,0 +1,219 @@
+import { Context } from '@nuxt/types';
+import { PluginInjection } from '~/types/utils';
+
+declare module '@nuxt/types' {
+  interface Context {
+    $browser: BrowserDetector;
+  }
+
+  interface NuxtAppOptions {
+    $browser: BrowserDetector;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $browser: BrowserDetector;
+  }
+}
+
+/**
+ * Utilities to detect the browser and get information on the current environment
+ * Based on https://github.com/google/shaka-player/blob/master/lib/util/platform.js
+ *
+ * @class BrowserDetector
+ */
+class BrowserDetector {
+  /**
+   * Check if the user agent of the navigator contains a key.
+   *
+   * @private
+   * @static
+   * @param {string} key - Key for which to perform a check.
+   * @returns {boolean}
+   * @memberof BrowserDetector
+   */
+  private userAgentContains(key: string) {
+    const userAgent = navigator.userAgent || '';
+    return userAgent.includes(key);
+  }
+
+  /* Desktop Browsers */
+
+  /**
+   * Check if the current platform is Mozilla Firefox.
+   *
+   * @returns
+   * @memberof BrowserDetector
+   */
+  isFirefox() {
+    return this.userAgentContains('Firefox/');
+  }
+
+  /**
+   * Check if the current platform is Microsoft Edge.
+   *
+   * @static
+   * @returns {boolean}
+   * @memberof BrowserDetector
+   */
+  isEdge() {
+    return this.userAgentContains('Edge/');
+  }
+
+  /**
+   * Check if the current platform is Google Chrome.
+   *
+   * @returns
+   * @memberof BrowserDetector
+   */
+  isChrome() {
+    // The Edge user agent will also contain the "Chrome" keyword, so we need
+    // to make sure this is not Edge.
+    return this.userAgentContains('Chrome') && !this.isEdge();
+  }
```

Comment:
What about chromium edge. Will that be classed as isEdge or isChrome?

Reply:
Given the lack of support for the Xbox client (we pulled it from the store) and the upcoming release of the Xbox One Series X/S in 2 months, I think it's safe to drop Edge 18 from our support.

As such, `isEdge()` will match on Edge Chromium, while `isChrome()` will match on Chromium-based browsers that are not Edge.

Reply:
Ok, that makes sense.